lib/open_nebula/resource_document_pool.rb

```diff
@@ -11,5 +11,13 @@ def factory(element_xml)
       s_template.load_body
       s_template
     end
+
+    def empty_document
+      ResourceDocument.new(ResourceDocument.build_xml, @client)
+    end
+
+    def include_id?(resource_document_id)
```

Comment:
Since you are matching `rd['NAME']`, the method should probably be called `include_name?`.